reconcile/utils/oc.py

```diff
@@ -1138,3 +1152,42 @@ def __init__(self, log_level, message):
 
     def __bool__(self):
         return False
+
+
+def validate_labels(labels) -> Iterable[str]:
+    """
+    Validate a label key/value against some rules from
+    https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set
+    """
+    if labels is None:
+        return []
+
+    err = []
+    for k, v in labels.items():
+        if len(v) > 63:
+            err.append(f'Label value longer than 63 chars: {v}')
+        pattern = '^[A-Za-z0-9][A-Za-z0-9-_\\.]{0,61}[A-Za-z0-9]$'
```

Comment:
Can you compile this regex outside the loop? It'll be easier to follow later on.